modules/sdk-core/src/bitgo/staking/goStakingWallet.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * @prettier
+ */
+import { BitGoBase } from '../bitgoBase';
+import { GetUserPrvOptions, IWallet } from '../wallet';
+import {
+  BaseGoStakeOptions,
+  FrontTransferSendRequest,
+  GoStakeFinalizeOptions,
+  GoStakeOptions,
+  GoStakingRequest,
+  IGoStakingWallet,
+  UnsignedGoStakingRequest,
+} from './iGoStakingWallet';
+import assert from 'assert';
+
+export class GoStakingWallet implements IGoStakingWallet {
+  private readonly bitgo: BitGoBase;
+
+  public wallet: IWallet;
+
+  constructor(wallet: IWallet) {
+    this.wallet = wallet;
+    this.bitgo = wallet.bitgo;
+  }
+
+  get accountId(): string {
+    return this.wallet.id();
+  }
+
+  async stake(coin: string, options: GoStakeOptions): Promise<GoStakingRequest> {
+    // call preview
+    const preview = await this.previewStake(coin, {
+      amount: options.amount,
+      clientId: options.clientId,
+    } as BaseGoStakeOptions);
+
+    // get keychain from wallet
+    const keychain = await this.wallet.baseCoin.keychains().getKeysForSigning({
+      wallet: this.wallet,
+    });
+
+    // NOTE: using this.wallet.prebuildAndSignTransaction did not work as expected because on signing step the coin was OFC instead of OFC token
+    // setup ofc coin to sign the transaction
+    const halfSignedTransaction = (await this.bitgo.coin(`ofc${coin}`).signTransaction({
+      txPrebuild: {
+        payload: preview.payload,
+      },
+      prv: this.wallet.getUserPrv({
+        keychain: keychain[0],
+        walletPassphrase: options.walletPassphrase,
+      } as GetUserPrvOptions),
```

Comment:
don't think the `as GetUserPrvOptions` is necessary, best to avoid unnecessary type casting like this

Reply:
updated